File: spring-boot-rest-integrated-demo/src/main/java/com/github/jccode/springbootrestintegrateddemo/repo/UserService.java
package com.github.jccode.springbootrestintegrateddemo.repo;

import com.github.jccode.springbootrestintegrateddemo.dao.UserMapper;
import com.github.jccode.springbootrestintegrateddemo.model.User;
import com.github.jccode.springbootrestintegrateddemo.model.UserCriteria;
import com.github.jccode.springbootsample.core.dao.CrudMapper;
import com.github.jccode.springbootsample.core.service.CrudServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService extends CrudServiceImpl<User> {

    @Autowired
    private UserMapper mapper;

    @Autowired
    public UserService(CrudMapper<User> mapper) {
        super(mapper);
    }

}

File: spring-boot-rest-integrated-demo/src/main/scala/com/github/jccode/springbootrestintegrateddemo2/service/UserService.scala
package com.github.jccode.springbootrestintegrateddemo2.service

import com.github.jccode.springbootrestintegrateddemo2.model.{User, UserCriteria}
import com.github.jccode.springbootrestintegrateddemo2.repo.UserMapper
import com.github.jccode.springbootsample.core.service.CrudServiceImpl
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import scala.collection.JavaConverters._

@Service
class UserService(@Autowired override val mapper: UserMapper[User]) extends CrudServiceImpl[User](mapper) {

  def count: Int = mapper.countByExample(new UserCriteria).toInt

  def findByName(name: String): List[User] = {
    val criteria: UserCriteria = new UserCriteria
    criteria.createCriteria.andNameEqualTo(name)
    mapper.selectByExample(criteria).asScala.toList
  }
}
